src/pyodide/internal/python.ts

```diff
@@ -44,6 +44,7 @@ import { default as MetadataReader } from 'pyodide-internal:runtime-generated/me
 function prepareWasmLinearMemory(Module: Module): void {
   // Note: if we are restoring from a snapshot, runtime is not initialized yet.
   Module.noInitialRun = !SHOULD_RESTORE_SNAPSHOT;
+  Module.Function = Function;
```

Comment:
Why not use Module.globalThis.Function instead of defining another member on Module?